packages/atomic/src/components/commerce/atomic-commerce-pager/e2e/atomic-commerce-pager.e2e.ts

```diff
@@ -0,0 +1,232 @@
+      await expect(pager.numericButton(3)).toHaveAttribute(
+        'part',
+        expect.stringContaining('active-page-button')
+      );
+    });
+
+    test('pager button 1 should not be selected', async ({pager}) => {
+      await expect(pager.numericButton(1)).not.toHaveAttribute(
+        'part',
+        'active-page-button'
+      );
+    });
+
+    test('should include the page in the hash', async ({page}) => {
+      await expect(page.url()).toContain('#page=2');
+    });
+
+    test('nextButton should be enabled', async ({pager}) => {
+      await expect(pager.nextButton).toBeEnabled();
+    });
+    test('previousButton should be enabled', async ({pager}) => {
+      await expect(pager.previousButton).toBeEnabled();
+    });
+  });
+
+  test.describe('after clicking on page 5', () => {
+    test.beforeEach(async ({pager}) => {
+      await pager.numericButton(5).click();
+    });
+
+    test('pager button 5 should not be selected', async ({pager}) => {
+      await expect(pager.numericButton(5)).not.toHaveAttribute(
+        'part',
+        'active-page-button'
+      );
+    });
+  });
+});
+
+test.describe('with a valid page in the hash', () => {
+  test.beforeEach(async ({page}) => {
+    await page.goto(
+      'http://localhost:4400/iframe.html?id=atomic-commerce-pager--default&viewMode=story#page=2'
+    );
+  });
+
+  test('pager button 3 should be selected', async ({pager}) => {
+    await expect(pager.numericButton(3)).toHaveAttribute(
+      'part',
+      expect.stringContaining('active-page-button')
+    );
+  });
+
+  test('should be A11y compliant', async ({pager, makeAxeBuilder}) => {
+    await pager.hydrated.waitFor();
+    const accessibilityResults = await makeAxeBuilder().analyze();
+    expect(accessibilityResults.violations).toEqual([]);
+  });
+});
+
+test.describe('with an invalid page in the hash', () => {
+  test.beforeEach(async ({page}) => {
+    await page.goto(
+      'http://localhost:4400/iframe.html?id=atomic-commerce-pager--default&viewMode=story#page=8k3'
+    );
+  });
+
+  test('pager button 9 should be selected', async ({pager}) => {
+    await expect(pager.numericButton(9)).toHaveAttribute(
+      'part',
+      expect.stringContaining('active-page-button')
+    );
+  });
+
+  test('should be A11y compliant', async ({pager, makeAxeBuilder}) => {
+    await pager.hydrated.waitFor();
+    const accessibilityResults = await makeAxeBuilder().analyze();
+    expect(accessibilityResults.violations).toEqual([]);
+  });
+});
+
+test.describe('with number-of-pages=3', () => {
+  test.beforeEach(async ({page}) => {
+    await page.goto(
+      'http://localhost:4400/iframe.html?id=atomic-commerce-pager--default&viewMode=story&args=attributes-number-of-pages:3'
+    );
+  });
+
+  test('should display 3 pages', async ({pager}) => {
+    await expect(pager.pages).toHaveCount(3);
+  });
+});
+
+test.describe('with numberOfPages=-5', () => {
```

Comment:
Not really related to this component directly, but I noticed that on first load of a commerce search page, with an invalid value for page or number-of-pages, the interface fails to load..!

Reply:
Isn't it a bad experience for the user, let's say they use a url that has an invalid value for whatever reason, the search interface will not load and they may just leave thinking the search is broken. I don't think it will happen often but it makes more sense to me to just ignore the invalid value instead of preventing the interface from loading..